domain/machine/state/state.go

```diff
@@ -183,5 +181,38 @@ func (st *State) GetMachineLife(ctx context.Context, machineId string) (*life.Li
 
 		return nil
 	})
+
+	if err != nil && errors.Is(err, sqlair.ErrNoRows) {
+		return nil, errors.NotFoundf("machine %q", machineId)
+	}
+
 	return &lifeResult, errors.Annotatef(err, "getting life status for machines %q", machineId)
 }
+
+// ListAllMachines retrieves the ids of all machines in the model.
+func (st *State) ListAllMachines(ctx context.Context) ([]coremachine.ID, error) {
```

Comment:
In the `state` packages, I would tend to use very mechanical names, so this would be `AllMachineNames`.